### PR TITLE
Fix: hardcode better SDK name and trim version to valid SemVer format

### PR DESCRIPTION
**Pull request by typotter**
fixes FF-2940

Also plumbs through app details as a dict, including tests

### CODE REVIEW

File: dot-net-sdk/helpers/AppDetails.cs [hunk 1]
```diff
@@ -4,47 +4,51 @@ namespace eppo_sdk.helpers;
 
 public class AppDetails
 {
-    private const string SDK_LANG = "c#";
-    static AppDetails? _instance;
+    static AppDetails? instance;
```
Comment (typotter) [inline dot-net-sdk/helpers/AppDetails.cs:7]: conforming to c# convention

File: dot-net-sdk/helpers/AppDetails.cs [hunk 2]
```diff
@@ -4,47 +4,51 @@ namespace eppo_sdk.helpers;
...
-        this._version = Assembly.GetExecutingAssembly().GetName().Version?.ToString();
-        this._name = Assembly.GetExecutingAssembly().GetName().Name;
+        // .net returns a 4-segmented version string (MAJOR.MINOR.BUILD.REVISION) here but we want to stick to semver standards (3-segment).
+        // We use a convention of Major.Minor.Patch when setting the package version; dotnet parses this to Major.Minor.Build and apprends
+        // the `.0` for Revision automatically. We can safely ignore it.
+        var fullVersion = Assembly.GetExecutingAssembly().GetName().Version!;
+        version = $"{fullVersion.Major}.{fullVersion.Minor}.{fullVersion.Build}";
+        
+        // Hardcoded for now; update soon with client/server split.
+        name = "dotnet-server";
```
Comment (typotter) [inline dot-net-sdk/helpers/AppDetails.cs:34]: new SDK name.

File: dot-net-sdk/helpers/AppDetails.cs [hunk 3]
```diff
@@ -4,47 +4,51 @@ namespace eppo_sdk.helpers;
 
 public class AppDetails
 {
-    private const string SDK_LANG = "c#";
```
Comment (typotter) [inline dot-net-sdk/helpers/AppDetails.cs:7]: dropping sdkLanguage as it's not logged anyway.

Comment (schmit) [reply dot-net-sdk/helpers/AppDetails.cs:7]: FWIW, In other SDKs we log it as part of the metadata in the assignment event.

Comment (schmit) [reply dot-net-sdk/helpers/AppDetails.cs:7]: Never mind I'm commenting too quickly as usual

**Review by schmit — APPROVED** [review]
👌

File: dot-net-sdk/helpers/AppDetails.cs [hunk 4]
```diff
@@ -4,47 +4,51 @@ namespace eppo_sdk.helpers;
...
+        return instance;
     }
 
     private AppDetails()
     {
-        this._version = Assembly.GetExecutingAssembly().GetName().Version?.ToString();
-        this._name = Assembly.GetExecutingAssembly().GetName().Name;
+        // .net returns a 4-segmented version string (MAJOR.MINOR.BUILD.REVISION) here but we want to stick to semver standards (3-segment).
+        // We use a convention of Major.Minor.Patch when setting the package version; dotnet parses this to Major.Minor.Build and apprends
+        // the `.0` for Revision automatically. We can safely ignore it.
```
Comment (leoromanovsky) [inline dot-net-sdk/helpers/AppDetails.cs:29]: thank you for the detailed comment

**Review by leoromanovsky — APPROVED** [review]

File: dot-net-sdk/helpers/AppDetails.cs (hunk 3 above)
Comment (typotter) [reply dot-net-sdk/helpers/AppDetails.cs:7]: *Not logged by the CDN is what I should have said here.

**Comment by leoromanovsky** [comment]
Still lgtm!

**Pull request merged**